### PR TITLE
Plugins: Allow using full plugin ID as data source API group name

### PR DESCRIPTION
**Pull request by wbrowne**
**What is this feature?**

Makes it possible to use the full plugin ID as the API group name for data source API servers.

**Why do we need this feature?**

This allows us to start to work with both core and non-Core Grafana authored data source plugins. This likely isn't the final thing we will land on but it keeps things simple for now.

**NOTE:** Alternatively, we keep the functionality as is, and introduce a parameter/config so that we know the explicit plugin ID we are dealing with.

### CODE REVIEW

**Review by andresmgot — COMMENTED** [review]
I'd just stop doing this, at the end, we [need to have a map](https://github.com/grafana/grafana/blob/main/pkg/services/apiserver/standalone/factory.go#L97) between this group name and the plugin ID so we can just use the pluginID always to simplify our lifes (even if it's ugly)

**Review by marefr — APPROVED** [review]
I agree with @andresmgot lets make our life easier, but this LGTM for now

**Comment by ryantxu** [comment]
I agree it is a bit of a pain that there is not a 1:1 mapping -- big picture we need to decide what the most appropriate group names are and how that can live nicely in a k8s first world.  We currently have a strategy that lets us map pluginID > group, but the other way is not deterministic.

Eventually I think the group name (and apiVersion) should be part of the plugin.json -- I'm fearful that if we make a 1:1 mapping now, then that will get encoded in ways that are hard to undo, and I am not yet willing to tell the world that our new naming strategy is: 😬 
```
org-name-datasource.datasource.grafana.app
```

File: pkg/plugins/apiserver_test.go [hunk 1]
```diff
@@ -7,15 +7,16 @@ import (
...
 
 func TestUtils(t *testing.T) {
-	// multiple flavors of the same idea
-	require.Equal(t, "tempo.datasource.grafana.app", getIDIgnoreError("tempo"))
-	require.Equal(t, "tempo.datasource.grafana.app", getIDIgnoreError("grafana-tempo-datasource"))
-	require.Equal(t, "tempo.datasource.grafana.app", getIDIgnoreError("tempo-datasource"))
+	// all other datasources use plugin ID as prefix (as long as is valid)
+	require.Equal(t, "name.datasource.grafana.app", getIDIgnoreError("name"), "single word is accepted")
+	require.Equal(t, "name-datasource.datasource.grafana.app", getIDIgnoreError("name-datasource"), "multiple words are accepted with '-datasource' suffix")
+	require.Error(t, getErrorIgnoreValue("org-name"), "multiple words are not accepted without '-datasource' suffix")
```
Comment (andresmgot) [inline pkg/plugins/apiserver_test.go:13]: I don't know if it matters but we could have `azure-monitor`, which is just the name but it includes a `-`.

**Review by andresmgot — APPROVED** [review]
I think we are unnecessarily complicating this but feel free to move forward with this

**Comment by andresmgot** [comment]
> and I am not yet willing to tell the world that our new naming strategy is: 😬
> 
> ```
> org-name-datasource.datasource.grafana.app
> ```

I don't think that matters if it's really internal

**Comment by marefr** [comment]
> I'm fearful that if we make a 1:1 mapping now, then that will get encoded in ways that are hard to undo, and I am not yet willing to tell the world that our new naming strategy is: 😬 
> 
> ```
> 
> org-name-datasource.datasource.grafana.app
> 
> ```

@ryantxu I hear you but don't still get why not. It's the simplest thing we can do currently and it will be understood by everyone, unique, without doing any conversion for consumers. 

If introducing things in plugin.json we can continue to support both if needed. Problem with this approach is that it will not be guaranteed to be unique, hence why I suggest the simplest first approach.

**Comment by ryantxu** [comment]
> I don't think that matters if it's really internal

It is *not* internal -- for alerting it is, but um; remember that that is just our first example.  When this works, this will be *the* way to expose datasources: `/api/datasource/*` will be removed

**Comment by ryantxu** [comment]
Currently it is only internal that we need to deal with a mapping -- if we encode 1:1 now, we won't be able to revisit that easily later.  

Perhaps my hidden agenda here is that any k8s api should eventually be able to be a valid datasource if it implements the right things.  The plugins catalog/registry should not control what is a plugin -- rather api discovery etc.  With a 1:1 mapping, our internal teams will likely skip this and continue to depend on plugin registry as the source of truth rather than apiDiscovery

**Pull request closed without merging**